src/tangent_types/thunks.jl

```diff
@@ -141,7 +145,11 @@ macro thunk(body)
     # Basically `:(Thunk(() -> $(esc(body))))` but use the location where it is defined.
     # so we get useful stack traces if it errors.
     func = Expr(:->, Expr(:tuple), Expr(:block, __source__, body))
-    return :(Thunk($(esc(func))))
+    return quote
+        $(esc(_usethunks))() ?
+            Thunk($(esc(func))) :
+            $(esc(func))()
+    end
```

Comment:
Comments on this approach over `:(_usethunks() ? Thunk($(esc(func))) : $(esc(body)))`, from https://github.com/JuliaDiff/ChainRulesCore.jl/pull/568? My hope there was that including the body directly, instead of making and calling a function, might be slightly simpler for e.g. Zygote to understand.